### DISCUSSION TITLE
Query stuck in fetching when invalidated

### DISCUSSION THREAD

**Discussion by agos** [TanStack/query#5234] [Q&A]
Hi,

a weird thing is happening to one of my queries: when I invalidate it (via `queryClient` or developer tools) it gets refetched (network calls starts, succeeds, function returns value) but the query state stays in `fetching`, and the new data never makes it to the query or its consumers. I have many similar queries that don't exhibit the same behavior.

Has something like that happened to anyone? any clues on where to look? for now I'm resetting the query as a workaround but it's clearly less than optimal

**Comment by TkDodo** [comment]
can you show a codesandbox reproduction please?
